angular/projects/admin-nrpti/src/app/records/agreements/agreement-lng-detail/agreement-lng-detail.component.ts

```diff
@@ -107,6 +108,10 @@ export class AgreementLNGDetailComponent extends RecordComponent implements OnIn
     return this.data && this.data.read && this.data.read.includes('public');
   }
 
+  userInLngRole() {
```

Comment:
Starting to see this and other functions repeated a fair amount. Wonder if it might be worth moving into a helper file, or maybe creating an interface that we could implement on all components that need auth access?

Reply:
Yeah, thought about moving everything into a common and sharing the methods. I moved the consts for roles into common record, but we probably don't need these methods shared for lng and nrced public. I think I'll move them into factory as all screens will have that available, along with the main role check.

Reply:
For now I've moved the `userIn...` helper methods into the factoryService (as it's the facade to all NRCED services) alongside the role check method. This should get rid of each nrced/lng page implementing the call themselves. They're now just calling `this.factoryService.userIn...`.

If we need the authentication for any other application in the future, we can move all of this (along with the keycloak stuff) into a common auth interface.